### ISSUE TITLE
RootedTransactioner: packfile is not added to siva file.

### ISSUE THREAD

**Issue by ajnavarro** [src-d/core-retrieval#16]
After go-billy.v3 migration, packfiles are not copied into the siva file. This behavior doesn't cause any error, just  the packfile is not there.

**Comment by alcortesm** [comment]
@mcuadros is proposing a fix to this problem, in this PR from @ajnavarro : https://github.com/src-d/go-billy-siva/pull/9.

It has no tests and the problem or the solution are not explained, but presumably it creates a filesystem that writes all files to a siva medium that ignores files under the /tmp dir.  It is not clear what role the sync part has in all this.

The task here will involve:
- [ ] (2 hours) write a test that reproduces the problem
- [ ] (3 hours) understand the proposed solution and explain it
- [ ] (0 hours) see if it passes the test
- [ ] (2 hours) adapting borges to this new version of go-billy-siva (involves glide)

**Comment by alcortesm** [comment]
@ajnavarro proposes a test in https://github.com/src-d/core-retrieval/pull/17.

**Comment by alcortesm** [comment]
This has been solved in borges master brach commit: aae7c98668c8adf79481728ea81cf0263a4510f9.

I tested it using https://github.com/alcortesm/demo-borges commit a693c3f3b49192c1c62d9d9d7985113a34f96bea.

<details>
This is how to reproduce the check:

```
; ./run-borges.bash 
[borges] compiling and installing...
Makefile.main:72: target 'github.com/Masterminds/glide' given more than once in the same rule
Makefile:32: warning: overriding recipe for target 'github.com/Masterminds/glide'
Makefile.main:73: warning: ignoring old recipe for target 'github.com/Masterminds/glide'
Makefile:31: target 'github.com/Masterminds/glide' given more than once in the same rule
Makefile:35: warning: overriding recipe for target '/home/alberto/room/gospace/src/github.com/src-d/borges/vendor'
Makefile.main:76: warning: ignoring old recipe for target '/home/alberto/room/gospace/src/github.com/src-d/borges/vendor'
Makefile:38: warning: overriding recipe for target 'github.com/src-d/borges'
Makefile.main:79: warning: ignoring old recipe for target 'github.com/src-d/borges'
Makefile:41: warning: overriding recipe for target 'test'
Makefile.main:147: warning: ignoring old recipe for target 'test'
Makefile:44: warning: overriding recipe for target 'test-coverage'
Makefile.main:150: warning: ignoring old recipe for target 'test-coverage'
borges_darwin_amd64/
borges_darwin_amd64/borges
borges_linux_amd64/
borges_linux_amd64/borges
[borges-postgres] is runnning from a previous execution, stopping it...
[borges-postgres] found from previous execution, removing it...
[borges-postgres] running docker container...
[borges-rabbitmq] is runnning from a previous execution, stopping it...
[borges-rabbitmq] found from previous execution, removing it...
[borges-rabbitmq] running docker container...
waiting for containers to boot up...
[borges-postgres] creating tables...
deleting archive from previous executions...
[borges] configuring environment...
[borges] running producer...
DBUG[07-03|12:41:12] starting                                 module=producer caller=producer.go:54
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34b5-9ba8-a378-0607cec30214 caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34bb-0056-aedf-3579a4fbae0b caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34c2-b895-85b7-2a5d3c5d117d caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34c6-3cc8-3fb8-a076db1ec532 caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34cb-2df6-a8c2-15a1ad02229a caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34d1-54e9-56c4-dde0e3806679 caller=producer.go:72
INFO[07-03|12:41:12] job queued                               module=borges RepositoryID=015d080a-34d8-41fa-0b81-151381419d75 caller=producer.go:72
EROR[07-03|12:41:12] error obtaining next job                 module=producer err="expected absolute URL: " caller=producer.go:71
DBUG[07-03|12:41:12] stopping                                 module=producer caller=producer.go:80
[borges] configuring environment...
[borges] running consumer...
DBUG[07-03|12:41:14] starting                                 module=worker id=0 caller=worker.go:33
DBUG[07-03|12:41:14] starting                                 module=worker id=1 caller=worker.go:33
DBUG[07-03|12:41:14] job started                              module=borges WorkerID=0 RepositoryID=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb caller=consumer.go:44
DBUG[07-03|12:41:14] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34b5-9ba8-a378-0607cec30214 caller=consumer.go:44
DBUG[07-03|12:41:14] repository model obtained                job=015d080a-34b5-9ba8-a378-0607cec30214 status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:14] endpoint selected                        job=015d080a-34b5-9ba8-a378-0607cec30214 endpoint=git://github.com/alcortesm/tgz.git caller=archiver.go:100
DBUG[07-03|12:41:14] local temporary directory created        job=015d080a-34b5-9ba8-a378-0607cec30214 temp-path=local_repos/015d080a-34b5-9ba8-a378-0607cec30214/1499078474940505497 caller=archiver.go:109
DBUG[07-03|12:41:14] local repository created                 job=015d080a-34b5-9ba8-a378-0607cec30214 caller=archiver.go:116
DBUG[07-03|12:41:14] repository model obtained                job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:14] endpoint selected                        job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb endpoint=git://github.com/alcortesm/MazeGame.git caller=archiver.go:100
DBUG[07-03|12:41:14] local temporary directory created        job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb temp-path=local_repos/015d080a-34af-7ba6-26e3-9a6ab4c2fdcb/1499078474942900823 caller=archiver.go:109
DBUG[07-03|12:41:14] local repository created                 job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb caller=archiver.go:116
DBUG[07-03|12:41:15] changes obtained                         job=015d080a-34b5-9ba8-a378-0607cec30214 roots=1 caller=archiver.go:143
DBUG[07-03|12:41:15] repository processed                     job=015d080a-34b5-9ba8-a378-0607cec30214 caller=archiver.go:148
INFO[07-03|12:41:15] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34b5-9ba8-a378-0607cec30214 caller=consumer.go:51
DBUG[07-03|12:41:15] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34bb-0056-aedf-3579a4fbae0b caller=consumer.go:44
DBUG[07-03|12:41:15] repository model obtained                job=015d080a-34bb-0056-aedf-3579a4fbae0b status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:15] endpoint selected                        job=015d080a-34bb-0056-aedf-3579a4fbae0b endpoint=git://github.com/alcortesm/sample.git caller=archiver.go:100
DBUG[07-03|12:41:15] local temporary directory created        job=015d080a-34bb-0056-aedf-3579a4fbae0b temp-path=local_repos/015d080a-34bb-0056-aedf-3579a4fbae0b/1499078475638588333 caller=archiver.go:109
DBUG[07-03|12:41:15] local repository created                 job=015d080a-34bb-0056-aedf-3579a4fbae0b caller=archiver.go:116
DBUG[07-03|12:41:15] changes obtained                         job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb roots=1 caller=archiver.go:143
DBUG[07-03|12:41:16] changes obtained                         job=015d080a-34bb-0056-aedf-3579a4fbae0b roots=1 caller=archiver.go:143
DBUG[07-03|12:41:16] repository processed                     job=015d080a-34bb-0056-aedf-3579a4fbae0b caller=archiver.go:148
INFO[07-03|12:41:16] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34bb-0056-aedf-3579a4fbae0b caller=consumer.go:51
DBUG[07-03|12:41:16] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34c2-b895-85b7-2a5d3c5d117d caller=consumer.go:44
DBUG[07-03|12:41:16] repository model obtained                job=015d080a-34c2-b895-85b7-2a5d3c5d117d status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:16] endpoint selected                        job=015d080a-34c2-b895-85b7-2a5d3c5d117d endpoint=git://github.com/alcortesm/queue.git caller=archiver.go:100
DBUG[07-03|12:41:16] local temporary directory created        job=015d080a-34c2-b895-85b7-2a5d3c5d117d temp-path=local_repos/015d080a-34c2-b895-85b7-2a5d3c5d117d/1499078476335376478 caller=archiver.go:109
DBUG[07-03|12:41:16] local repository created                 job=015d080a-34c2-b895-85b7-2a5d3c5d117d caller=archiver.go:116
DBUG[07-03|12:41:17] changes obtained                         job=015d080a-34c2-b895-85b7-2a5d3c5d117d roots=1 caller=archiver.go:143
DBUG[07-03|12:41:17] repository processed                     job=015d080a-34c2-b895-85b7-2a5d3c5d117d caller=archiver.go:148
INFO[07-03|12:41:17] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34c2-b895-85b7-2a5d3c5d117d caller=consumer.go:51
DBUG[07-03|12:41:17] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34c6-3cc8-3fb8-a076db1ec532 caller=consumer.go:44
DBUG[07-03|12:41:17] repository model obtained                job=015d080a-34c6-3cc8-3fb8-a076db1ec532 status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:17] endpoint selected                        job=015d080a-34c6-3cc8-3fb8-a076db1ec532 endpoint=git://github.com/alcortesm/msgbroker.git caller=archiver.go:100
DBUG[07-03|12:41:17] local temporary directory created        job=015d080a-34c6-3cc8-3fb8-a076db1ec532 temp-path=local_repos/015d080a-34c6-3cc8-3fb8-a076db1ec532/1499078477189043836 caller=archiver.go:109
DBUG[07-03|12:41:17] local repository created                 job=015d080a-34c6-3cc8-3fb8-a076db1ec532 caller=archiver.go:116
DBUG[07-03|12:41:17] repository processed                     job=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb caller=archiver.go:148
INFO[07-03|12:41:17] job done                                 module=borges WorkerID=0 RepositoryID=015d080a-34af-7ba6-26e3-9a6ab4c2fdcb caller=consumer.go:51
DBUG[07-03|12:41:17] job started                              module=borges WorkerID=0 RepositoryID=015d080a-34cb-2df6-a8c2-15a1ad02229a caller=consumer.go:44
DBUG[07-03|12:41:17] repository model obtained                job=015d080a-34cb-2df6-a8c2-15a1ad02229a status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:17] endpoint selected                        job=015d080a-34cb-2df6-a8c2-15a1ad02229a endpoint=git://github.com/alcortesm/sysmon.git caller=archiver.go:100
DBUG[07-03|12:41:17] local temporary directory created        job=015d080a-34cb-2df6-a8c2-15a1ad02229a temp-path=local_repos/015d080a-34cb-2df6-a8c2-15a1ad02229a/1499078477479938112 caller=archiver.go:109
DBUG[07-03|12:41:17] local repository created                 job=015d080a-34cb-2df6-a8c2-15a1ad02229a caller=archiver.go:116
DBUG[07-03|12:41:17] changes obtained                         job=015d080a-34c6-3cc8-3fb8-a076db1ec532 roots=1 caller=archiver.go:143
DBUG[07-03|12:41:17] repository processed                     job=015d080a-34c6-3cc8-3fb8-a076db1ec532 caller=archiver.go:148
INFO[07-03|12:41:17] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34c6-3cc8-3fb8-a076db1ec532 caller=consumer.go:51
DBUG[07-03|12:41:17] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34d1-54e9-56c4-dde0e3806679 caller=consumer.go:44
DBUG[07-03|12:41:17] repository model obtained                job=015d080a-34d1-54e9-56c4-dde0e3806679 status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:17] endpoint selected                        job=015d080a-34d1-54e9-56c4-dde0e3806679 endpoint=git://github.com/alcortesm/demo-borges.git caller=archiver.go:100
DBUG[07-03|12:41:17] local temporary directory created        job=015d080a-34d1-54e9-56c4-dde0e3806679 temp-path=local_repos/015d080a-34d1-54e9-56c4-dde0e3806679/1499078477573624753 caller=archiver.go:109
DBUG[07-03|12:41:17] local repository created                 job=015d080a-34d1-54e9-56c4-dde0e3806679 caller=archiver.go:116
DBUG[07-03|12:41:17] changes obtained                         job=015d080a-34d1-54e9-56c4-dde0e3806679 roots=1 caller=archiver.go:143
DBUG[07-03|12:41:17] repository processed                     job=015d080a-34d1-54e9-56c4-dde0e3806679 caller=archiver.go:148
INFO[07-03|12:41:17] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34d1-54e9-56c4-dde0e3806679 caller=consumer.go:51
DBUG[07-03|12:41:17] job started                              module=borges WorkerID=1 RepositoryID=015d080a-34d8-41fa-0b81-151381419d75 caller=consumer.go:44
DBUG[07-03|12:41:17] repository model obtained                job=015d080a-34d8-41fa-0b81-151381419d75 status=pending last-fetch=nil references=0 caller=archiver.go:93
DBUG[07-03|12:41:17] endpoint selected                        job=015d080a-34d8-41fa-0b81-151381419d75 endpoint=git://github.com/alcortesm/demo-markdown.git caller=archiver.go:100
DBUG[07-03|12:41:17] local temporary directory created        job=015d080a-34d8-41fa-0b81-151381419d75 temp-path=local_repos/015d080a-34d8-41fa-0b81-151381419d75/1499078477939214745 caller=archiver.go:109
DBUG[07-03|12:41:17] local repository created                 job=015d080a-34d8-41fa-0b81-151381419d75 caller=archiver.go:116
DBUG[07-03|12:41:17] changes obtained                         job=015d080a-34cb-2df6-a8c2-15a1ad02229a roots=1 caller=archiver.go:143
DBUG[07-03|12:41:18] repository processed                     job=015d080a-34cb-2df6-a8c2-15a1ad02229a caller=archiver.go:148
INFO[07-03|12:41:18] job done                                 module=borges WorkerID=0 RepositoryID=015d080a-34cb-2df6-a8c2-15a1ad02229a caller=consumer.go:51
DBUG[07-03|12:41:18] changes obtained                         job=015d080a-34d8-41fa-0b81-151381419d75 roots=1 caller=archiver.go:143
DBUG[07-03|12:41:18] repository processed                     job=015d080a-34d8-41fa-0b81-151381419d75 caller=archiver.go:148
INFO[07-03|12:41:18] job done                                 module=borges WorkerID=1 RepositoryID=015d080a-34d8-41fa-0b81-151381419d75 caller=consumer.go:51
^C⏎                                                                                                                                                                                                                 [I] [12:41:21] alberto@cloak (master|✔) ~/r/g/s/g/a/demo-borges (130) 
; for file in /tmp/root-repositories/* ; siva list $file ; end
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 1.7 kB objects/pack/pack-0367e29975f1e69d7d342d48ba771de5d1686aa3.idx
-rw-rw-rw- Jul 03 12:41 4.4 kB objects/pack/pack-0367e29975f1e69d7d342d48ba771de5d1686aa3.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34d1-54e9-56c4-dde0e3806679
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 6.5 kB objects/pack/pack-6d3863a7f1e557c6d300b9101577c64b140ef4ae.idx
-rw-rw-rw- Jul 03 12:41  29 kB objects/pack/pack-6d3863a7f1e557c6d300b9101577c64b140ef4ae.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34cb-2df6-a8c2-15a1ad02229a
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 3.1 kB objects/pack/pack-d83a14b92266c45f9b8ffd846c58eaa493d25b66.idx
-rw-rw-rw- Jul 03 12:41  12 kB objects/pack/pack-d83a14b92266c45f9b8ffd846c58eaa493d25b66.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34c2-b895-85b7-2a5d3c5d117d
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41  34 kB objects/pack/pack-411240b3b55a283c60d4b4c452ced23b76fcd28f.idx
-rw-rw-rw- Jul 03 12:41 205 kB objects/pack/pack-411240b3b55a283c60d4b4c452ced23b76fcd28f.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34af-7ba6-26e3-9a6ab4c2fdcb
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 1.2 kB objects/pack/pack-4c4c63e0a8399f9a430d81c80f1dd44da0979822.idx
-rw-rw-rw- Jul 03 12:41  352 B objects/pack/pack-4c4c63e0a8399f9a430d81c80f1dd44da0979822.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34c6-3cc8-3fb8-a076db1ec532
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 2.8 kB objects/pack/pack-bf23ee318360ddd56077ddac9d216822d73a67b9.idx
-rw-rw-rw- Jul 03 12:41  11 kB objects/pack/pack-bf23ee318360ddd56077ddac9d216822d73a67b9.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34b5-9ba8-a378-0607cec30214
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 4.5 kB objects/pack/pack-cc48c28937076994b51a8575bfa3858206f0d344.idx
-rw-rw-rw- Jul 03 12:41  40 kB objects/pack/pack-cc48c28937076994b51a8575bfa3858206f0d344.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34bb-0056-aedf-3579a4fbae0b
-rw-rw-rw- Jul 03 12:41   41 B refs/tags/v1.0/015d080a-34bb-0056-aedf-3579a4fbae0b
-rw-rw-rw- Jul 03 12:41   23 B HEAD
-rw-rw-rw- Jul 03 12:41   20 B config
-rw-rw-rw- Jul 03 12:41 1.2 kB objects/pack/pack-6a19d6022661f68783276a62dec824cb1bfca82e.idx
-rw-rw-rw- Jul 03 12:41  327 B objects/pack/pack-6a19d6022661f68783276a62dec824cb1bfca82e.pack
-rw-rw-rw- Jul 03 12:41   41 B refs/heads/master/015d080a-34d8-41fa-0b81-151381419d75
```

</details>
